### PR TITLE
feat(admin-portal): connected accounts management on user detail page

### PR DESCRIPTION
**Pull request by mschilling**
## Summary
- Add user detail page (`/users/:id`) to the admin portal with connected accounts section
- Add CRUD API endpoints for connected accounts (`GET/POST/DELETE /users/:userId/connected-accounts`)
- Make user list rows clickable to navigate to the detail page
- Add `ConnectedAccountDto` shared type and `ConnectedAccounts` collection enum

## Test plan
- [x] API tested locally against emulators (GET/POST/DELETE + validation + 404 edge cases)
- [x] Build passes for `api`, `admin-portal`, `types`
- [ ] Deploy to test environment and verify UI flows (add/remove connected accounts)

### CODE REVIEW

File: apps/frontend/admin-portal/src/app/pages/users/user-detail.component.ts [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+import { ConnectedAccountDto, CONNECTED_ACCOUNT_PROVIDERS, UserDto } from '@codeheroes/types';
+import { UsersService } from '../../core/services/users.service';
+
+@Component({
+  selector: 'admin-user-detail',
+  standalone: true,
+  imports: [DatePipe, FormsModule, SuiButtonComponent],
+  template: `
+    <div>
+      <a class="back-link" (click)="goBack()">← Back to Users</a>
```
[the remark this answers is not in the document]
Comment (mschilling) [reply apps/frontend/admin-portal/src/app/pages/users/user-detail.component.ts]: Fixed: Changed `<a>` to `<button type="button">` with reset styles for proper keyboard and assistive tech support.

File: apps/frontend/admin-portal/src/app/pages/users/user-detail.component.ts [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+    this.#usersService.getConnectedAccounts(this.#userId).subscribe({
+      next: (accounts) => {
+        this.accounts.set(accounts);
+        this.accountsLoading.set(false);
+      },
+      error: () => {
+        this.accounts.set([]);
+        this.accountsLoading.set(false);
+      },
+    });
```
[the remark this answers is not in the document]
Comment (mschilling) [reply apps/frontend/admin-portal/src/app/pages/users/user-detail.component.ts]: Fixed: Added `accountsError` signal with dedicated error state UI and retry button, so failures are distinguishable from an empty list.
